From: Hendrik Brueckner <brueckner@linux.vnet.ibm.com>
Subject: s390/perf_events: compile only for CONFIG_64BIT
Patch-mainline: v3.7-rc1
Git-commit: 305e4f108b683a96254d36f916ecab79dcc3ea4f
References: bnc#900899, FATE#317698, LTC#110510

Summary:     s390/perf: add support for the CPU-measurement sampling facility
Description: Use the CPU-measurement sampling facility to obtain performance
             data for Linux in LPAR mode.  You can use the perf tool on Linux
             to access the hardware sample data of the CPU-measurement
             sampling facility.

Upstream-Description:

             s390/perf_events: compile only for CONFIG_64BIT

             The whole hardware support is only available in zArch mode.
             Fixes also this compile warning:

             arch/s390/kernel/perf_cpum_cf.c: In function ‘cpumf_pmu_init’:
             arch/s390/kernel/perf_cpum_cf.c:670:2: warning: left shift count >= width of type [enabled by default]

             Signed-off-by: Heiko Carstens <heiko.carstens@de.ibm.com>
             Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>


Signed-off-by: Hendrik Brueckner <brueckner@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 arch/s390/kernel/Makefile |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/arch/s390/kernel/Makefile
+++ b/arch/s390/kernel/Makefile
@@ -49,8 +49,11 @@ obj-$(CONFIG_DYNAMIC_FTRACE)	+= ftrace.o
 obj-$(CONFIG_FUNCTION_GRAPH_TRACER) += ftrace.o
 obj-$(CONFIG_FTRACE_SYSCALLS)  += ftrace.o
 obj-$(CONFIG_CRASH_DUMP)	+= crash_dump.o
+
+ifdef CONFIG_64BIT
 obj-$(CONFIG_PERF_EVENTS)	+= perf_event.o perf_cpum_cf.o
-obj-$(CONFIG_64BIT)		+= runtime_instr.o
+obj-y				+= runtime_instr.o
+endif
 
 # Kexec part
 S390_KEXEC_OBJS := machine_kexec.o crash.o
